### PR TITLE
feat(macros): Add lock comment functionality to modmacros

### PR DESCRIPTION
**Pull request by SpyTec**
Part of #94. Adds lock comment functionality to modmacros

`lockThread()` has been renamed to `lock()`
`unlockThread()` has been renamed to `unlock()`
`lockThread()` now points to `lock()`
`unlockThread()` now points to `unlock()`

## Creating modmacro
image

## Editing modmacro
image

## Triggering macro
image

## Result
image

### CODE REVIEW

File: extension/data/tbutils.js [hunk 1]
```diff
@@ -2387,7 +2387,7 @@ function initwrapper ({userDetails, newModSubs, cacheDetails}) {
                 });
         };
 
-        TBUtils.lockThread = function (id, callback) {
+        TBUtils.lock = function (id, callback) {
```
Comment (SpyTec) [inline extension/data/tbutils.js]: Should have `lockComment` and `lockThread? instead`

Comment (Geo1088) [reply extension/data/tbutils.js]: Since it's only one method in the API, it makes sense to keep it as is.

**Comment by creesch** [comment]
I am thinking the wording in some areas might need some adjustments. 
In the creation/edit windows possibly change it to `lock macro comment` to make clear it is about that. 
When posting a macro I just noticed that it says `this post will be locked` which is also confusing might  change that to `Parent post will be locked`? @Geo1088 thoughts?

**Comment by SpyTec** [comment]
Or just `lock reply` or `lock removal reply` would work as well. Not sure I can see how `this post will be locked` would be confused with something, what do you think people would see it as if not the post?

**Comment by creesch** [comment]
`lock reply` would work yeah. 


`This post will be locked` took me a moment to parse as the other `This`   items in the screenshot apply to the macro. It isn't a big deal though as it it is what it currently states.

**Comment by SpyTec** [comment]
I wonder, should we also make it possible to lock the comment being removed? At the moment the macro only locks the reply

**Comment by creesch** [comment]
Actually... I just noticed something it does. `lockthread` already locks the parent and we never bothered to check if the parent is a post or comment so it works on both. So I suggest we do the following: 

- On the UI side of things in creation/editing rename it to `lock parent` 
- When applying the macro make it fill in `comment` or `post` in the UI depending on what the parent is.

**Comment by SpyTec** [comment]
Sounds good. I'll fix it up and commit it here

**Comment by SpyTec** [comment]
Fix up

**Pull request merged**